src/webgpu/shader/execution/expression/expression.ts

```diff
@@ -1065,39 +1060,47 @@ export function generateBinaryToU32Cases(params0s: number[], params1s: number[],
  * @param scalar scalar param
  * @param vector vector param (2, 3, or 4 elements)
  * @param op the op to apply to scalar and vector
+ * @param quantize function to quantize all values in vectors and scalars
+ * @param scalarize function to convert numbers to Scalars
  */
-function makeU32VectorBinaryToVectorCase(
+function makeScalarVectorBinaryToVectorCase(
   scalar: number,
   vector: number[],
-  op: BinaryOp
+  op: BinaryOp,
+  quantize: QuantizeFunc,
+  scalarize: ScalarBuilder
 ): Case | undefined {
```

Comment:
Most of the diffs in this change are just replicating what I did for u32. The only interesting bit is in this file, where I generalized these four functions to take a `QuantizeFunc` and a `ScalarBuilder` so that I can easily reuse them for the generate functions for both U32 and I32 below.

Reply:
:+1: